src/datachain/lib/dc.py

```diff
@@ -953,10 +953,22 @@ def _extend_to_data_model(self, method_name, *args, **kwargs):
 
     @resolve_columns
     def order_by(self, *args, descending: bool = False) -> "Self":
-        """Orders by specified set of signals.
+        """Orders by specified set of columns.
 
         Parameters:
             descending (bool): Whether to sort in descending order or not.
+
+        Example:
+            ```py
+            dc.order_by("similarity_score", descending=True).limit(10)
+            ```
+
+        Note:
+            Order is not guaranteed when steps are added after an `order_by` statement.
+            I.e. when using `from_dataset` an `order_by` statement should be used if
+            the order of the records in the chain is important.
+            Using `order_by` directly before `limit` will give expected results.
```

Comment:
I assume `collect`, `show`, `to_parquet` - are not considered as steps and will give also a "desired" output?

Reply:
`collect` - should but not guranteed (explanation in the original ticket https://github.com/iterative/datachain/issues/477)
`show` - yes
`to_parquet` - should but not guaranteed (uses `collect_flatten`)

Reply:
Do you mean adding methods like `collect_ordered` OR adding an explicit `order_by` param to those methods or patching those functions? Is there a specific use case we're targeting or are we just trying to avoid confusion?

Reply:
I think it's a basic expectation that if you do `order_by().collect()` that you are getting a sorted result? collect is a way to iterate on the results I guess.

I mean that `collect` probably should be implemented similar to show() - something like ``order_by().collect()` should be a single SQL query probably